include/clock_control.h

```diff
@@ -72,6 +132,68 @@ static inline int clock_control_off(struct device *dev,
 	return api->off(dev, sys);
 }
 
+/**
+ * @brief Request clock to start with notification when clock has been started.
+ *
+ * User can request delayed start by providing exact information when clock
+ * should be ready. Driver ensures that clock is ready before requested time.
+ * It is the driver responsibility to take into account clock startup time.
+ * When clock is already running user callback will be called from the context
+ * of the function call else it is called from other context (e.g. clock
+ * interrupt).
+ *
+ * @param dev 	   Device.
+ * @param sys	   A pointer to an opaque data representing the sub-system.
+ * @param data	   Data structure containing a callback that is called when
+ *		   action is performed. Structure content must be valid until
+ *		   clock is started and user callback is called. Can be NULL.
+ * @param deadline When (given in @ref k_cycle_get_32()) clock must be ready.
```

Comment:
How you are going to implement "deadline" feature for Nordic?
Also, if this is mandatory feature, how other platforms are supposed to handle this?

Reply:
On Nordic I plan to use PPI and RTC1 channel (system clock source).
The simplest way for handling that on any platform is to start clock immediately. That covers functionality since deadline is met. It's only about low power. Platforms focused on lower power will find the way. One way is to use system timer adjusted to system latency (e.g. starting clock N ms earlier than requested).

Reply:
So Clock Control will be bound to system clock.

Reply:
yes

Reply:
I have withdrawn `deadline` argument from async on. In offline discussions we agreed with @pizi-nordic  and @carlescufi that it will be difficult to implement it on other platforms than nordic. It would be in fact public api which would be nordic specific for now.

Ideally, we should have kernel API like `k_uptime_us_get_32`. Absolute, monotonic microseconds counter which could be used for deadline.